src/parquet/util/cpu-info.h

```diff
@@ -54,44 +52,26 @@ class CpuInfo {
   static void VerifyCpuRequirements();
 
   /// Returns all the flags for this cpu
-  static int64_t hardware_flags() {
-    DCHECK(initialized_);
-    return hardware_flags_;
-  }
+  static int64_t hardware_flags();
 
   /// Returns whether of not the cpu supports this flag
-  inline static bool IsSupported(int64_t flag) {
-    DCHECK(initialized_);
-    return (hardware_flags_ & flag) != 0;
-  }
+  static bool IsSupported(int64_t flag);
```

Comment:
I would say to remove the DCHECK, @majetideepak @xhochy do you have an opinion? whether the `CpuInfo` is initialized is something we can test elsewhere